dcos/marathon.py

```diff
@@ -130,6 +130,8 @@ class Client(object):
     """
 
     def __init__(self, marathon_url, timeout=http.DEFAULT_TIMEOUT):
+        if not marathon_url.endswith('/'):
+            marathon_url += '/'
         self._base_url = marathon_url
```

Comment:
I think we should just do a more global patch so we don't end up having to make another quick fix laster.

`/`.join() won't add a trailing slash. 

something like this will work: urllib.parse.urljoin(self._base_url, urllib.parse, urljoin(path, '/'))`